### PR TITLE
Add support for transform plugins on tunnels.

### LINKED ISSUE #10313: Support transform plugins on tunnels
There are cases where it is useful to apply transforms to all bytes in a tunnel.  Unlike the HTTP transforms, the tunnel transforms need to handle multiple exchanges between the user agent and the origin server.

This is related to issue #10307. You will need a tunnel start hook to register tunnel transform plugins.

### PR DESCRIPTION
**Pull request by shinrich**
This requires the tunnel start hook from PR #10308, so this PR is building on that branch. So far only the last commit is unique to this PR.

This closes #10313

### CODE REVIEW

**Comment by moonchen** [comment]
Waiting for conflict resolution.

**Comment by shinrich** [comment]
Rebased.  Looks like I made a test plugin but didn't actually make an autest to use it.  Will get that added tomorrow.

**Review by moonchen — CHANGES_REQUESTED** [review]

**Comment by moonchen** [comment]
Looks good!  Just one small nitpick.

**Review by moonchen — DISMISSED** [review]

**Comment by shinrich** [comment]
[approve ci autest]

File: tests/gold_tests/tunnel/tunnel_transform.test.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+path1 = tr.Processes.Default.Streams.stdout.AbsPath
+
+tr2 = Test.AddTestRun("Check the input bytes sent and fetch outptut bytes sent")
+tr2.Processes.Default.Command = 'traffic_ctl metric get tunnel_transform.os.bytes_sent'
+tr2.Processes.Default.ReturnCode = 0
+tr2.Processes.Default.Env = ts.Env
+tr2.StillRunningAfter = ts
+tr2.StillRunningAfter = server
+tr2.Processes.Default.Streams.stdout = Testers.Lambda(lambda info, tester: check_range(path1, 700, 800))
```
Comment (shinrich) [inline tests/gold_tests/tunnel/tunnel_transform.test.py:158]: I couldn't figure out how to get the current streams stdout fed into the check_range function.  I assume it is available via the info argument, but that appears to be a structure and I couldn't find the methods to get the data stream.  So I'm testing the output by reading the standard output file from the previous test case (a pattern I had used in some tls tests).  @dragon512 or @bneradt I'd appreciate pointers to clean this up.  Would be happy to do that in a later PR.  This logic is functional and verifies that the metric falls within a range.

**Review by moonchen — APPROVED** [review]

**Pull request merged**